### PR TITLE
feat: return nonce for frictionless cards

### PR DESCRIPTION
**Pull request by Nick-Zag**

### CODE REVIEW

File: packages/demand-bloc-payment/src/PaymentBloc.ts [hunk 1]
```diff
@@ -95,7 +95,10 @@ export class PaymentBloc {
         }
       }
 
-      const paymentNonce = await this.getPaymentDetails()
+      const { nonce: paymentNonce, resultCode } = await this.getPaymentDetails()
+      if (resultCode) {
+        return { ok: true, nonce: nonce }
```
Comment (ykarduba) [inline packages/demand-bloc-payment/src/PaymentBloc.ts:100]: `return { ok: true, nonce }`

File: packages/demand-bloc-payment/src/PaymentBloc.ts [hunk 2]
```diff
@@ -118,9 +121,9 @@ export class PaymentBloc {
 
   async getPaymentNonce(): Promise<PaymentNonceResponse> {
     try {
-      const nonce = await this.getPaymentDetails()
+      const { nonce } = await this.getPaymentDetails()
 
-      return { ok: true, nonce }
+      return { ok: true, nonce: nonce }
```
Comment (ykarduba) [inline packages/demand-bloc-payment/src/PaymentBloc.ts:126]: `return { ok: true, nonce }`

File: packages/demand-bloc-payment/src/providers/AdyenProvider.ts [hunk 3]
```diff
@@ -10,6 +10,7 @@ import {
   CompleteThreeDSecureVerificationParams,
   Payer,
   AdyenShopperData,
+  AUTHORISED,
```
Comment (juliamaksimchik) [inline packages/demand-bloc-payment/src/providers/AdyenProvider.ts:13]: could we do something to get ResultCodes.AUTHORISED in a condition ? Just AUTHORISED is not very clear

File: packages/demand-bloc-payment/src/providers/AdyenProvider.ts [hunk 4]
```diff
@@ -142,6 +143,11 @@ export class AdyenProvider implements Provider {
 
     handleRefusalResponse(makePaymentResponse.body.payload)
 
+    if (makePaymentResponse.body.payload.resultCode === AUTHORISED) {
+      this.nonce = makePaymentResponse.body.trip_id
+      return ['meta.trip_id', makePaymentResponse.body.trip_id, makePaymentResponse.body.payload.resultCode]
```
Comment (juliamaksimchik) [inline packages/demand-bloc-payment/src/providers/AdyenProvider.ts:148]: Why we need to return result code ?

Comment (Nick-Zag) [reply packages/demand-bloc-payment/src/providers/AdyenProvider.ts:148]: To distinct this case in PaymentBloc as in verifyCardWithThreeDSecure  method we use condition for returning resultNonce and in this case this condition doesn't work

**Review by ykarduba — APPROVED** [review]

**Review by andrei-oksentyk — APPROVED** [review]

**Pull request merged**